zmu_cortex_m/src/gdb/server.rs

```diff
@@ -0,0 +1,171 @@
+//!
+//! Flash Memory simulation
+//!
+//!
+
+use gdbstub::stub::run_blocking;
+use gdbstub::stub::SingleThreadStopReason;
+use gdbstub::conn::Connection;
+use gdbstub::conn::ConnectionExt;
+use gdbstub::target::Target;
+use gdbstub::common::Signal;
+use gdbstub::stub::GdbStub;
+use gdbstub::stub::DisconnectReason;
+
+use crate::{MemoryMapConfig};
+use crate::gdb::conn;
+use conn::TcpConnection;
+use crate::gdb::simulation::SimulationEvent;
+use crate::gdb::simulation::SimulationRunEvent;
+
+use crate::gdb::target::ZmuTarget;
+
+use crate::semihosting::SemihostingCommand;
+use crate::semihosting::SemihostingResponse;
+
+///
+/// The gdb Server
+/// 
+pub struct GdbServer {
+    // number: i32,
+    target: ZmuTarget,
+}
+
+/// 
+impl GdbServer {
+
+    ///
+    pub fn new(
+        code: &[u8],
+        semihost_func: Box<dyn FnMut(&SemihostingCommand) -> SemihostingResponse + 'static>,
+        map: Option<MemoryMapConfig>,
+        flash_size: usize,
+    ) -> Result<GdbServer, &'static str> {
+
+        let target = ZmuTarget::new(code, semihost_func, map, flash_size);
+
+        Ok(GdbServer {target})
+    }
+
+    ///
+    pub fn start(&mut self) -> Result<u32, &'static str> {
+        println!("Starting GDB server");
+        let mut exit_code = 0;
+        let conn = match conn::TcpConnection::new_localhost(9001) {
+            Ok(conn) => conn,
+            Err(e) => return Err(e),
+        };
+
+        let gdb = GdbStub::new(conn);
+
+        match gdb.run_blocking::<EventLoop>(&mut self.target) {
+            Ok(disconnect_reason) => match disconnect_reason {
+                DisconnectReason::Disconnect => {
+                    println!("GDB client has disconnected. Running to completion...");
```

Comment:
please change println! s to use correct **log** module macros

Reply:
done!